### ISSUE TITLE
Make commenting out code aware of analyzers

### ISSUE THREAD

**Issue by Forgind** [dotnet/roslyn#66741]
## Summary
<img width="461" alt="image" src="https://user-images.githubusercontent.com/12969783/217364827-2ca8a90a-9953-45c7-b966-6a40912c3a99.png">

I commented out some code using VS's built-in comment-out-code feature. It did comment out the code, but it styled it in a way that violated SA1005 (and SA1515, though I'm less worried about that one). I would propose that commenting out blocks of code should be aware of analyzers and automatically format things properly. This is particularly important when an analyzer is set to error (or warning with warn-as-error). In that case, I comment out code, try to build, fail, tweak, the comment, and progress. In the worst case, it can take me several builds to get it to work because I hit one analyzer warning/error after another. It would be nice if this were a seamless experience.

**Comment by Youssef1313** [comment]
SA1005 docs say:

> An exception to this rule occurs when the comment is being used to comment out a line of code. In this case, the space can be omitted if the comment begins with four forward slashes to indicate out-commented code.

**Comment by Forgind** [comment]
No. This is an issue with how VS comments code when you use the built-in feature. This should be reopened.

**Comment by CyrusNajmabadi** [comment]
@Forgind This is how we've commented out code for 25 years.  It's a normal style that the language and tooling has aligned on for all that time.  There is an issue here with an external analzyer and the personal rules they've decided.  Just as if there was an analyzer that didn't like the formatting of other constructs, or 3rd party analyzers that didn't like the *usage* *of certain constructs (say like 'lambdas' or 'ternary expressions'), that's out of scope for roslyn.  It is a non-goal to have 3rd party rules affect our own features.  

> This is an issue with how VS comments code when you use the built-in feature.

From our perspective, it is commenting out code as we expect and how we feel is appropriate.  We have no goal for external analyzers to impact how our features operate.
